pandas/io/formats/excel.py

```diff
@@ -356,7 +356,15 @@ def __init__(self, df, na_rep='', float_format=None, cols=None,
             self.styler = None
         self.df = df
         if cols is not None:
-            self.df = df.loc[:, cols]
+
+            # all missing, raise
+            if not len(Index(cols) & df.columns):
+                raise KeyError
+
+            # 1 missing is ok
+            # TODO(jreback) this should raise
+            # on *any* missing columns
+            self.df = df.reindex(columns=cols)
```

Comment:
This indeed currently works, but I think it is certainly OK to change that. Or maybe also raise futurewarning as with loc?